src/order/symm_diff.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2021 Bryan Gin-ge Chen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz, Bryan Gin-ge Chen
+-/
+
+import order.boolean_algebra
+
+/-!
+# Symmetric difference
+
+The symmetric difference or disjunctive union of sets `A` and `B` is the set of elements that are
+in either `A` or `B` but not both. Translated into propositions, the symmetric difference is `xor`.
+
+The symmetric difference operator (`symm_diff`) is defined in this file for any type with `⊔` and
+`\` via the formula `(A \ B) ⊔ (B \ A)`, however the theorems proved about it only hold for
+`generalized_boolean_algebra`s and `boolean_algebra`s.
+
+The symmetric difference is the addition operator in the Boolean ring structure on Boolean algebras.
+
+## Main declarations
+
+* `symm_diff`: the symmetric difference operator, defined as `(A \ B) ⊔ (B \ A)`
+
+In generalized Boolean algebras, the symmetric difference operator is:
+
+* `symm_diff_comm`: commutative, and
+* `symm_diff_assoc`: associative.
+
+## Notations
+
+* `a Δ b`: `symm_diff a b`
+
+## References
+
+The proof of associativity follows the note "Associativity of the Symmetric Difference of Sets: A
+Proof from the Book" by John McCuan:
+
+* <https://people.math.gatech.edu/~mccuan/courses/4317/symmetricdifference.pdf>
+
+## Tags
+boolean ring, generalized boolean algebra, boolean algebra, symmetric differences
+-/
+
+/-- The symmetric difference operator on a type with `⊔` and `\` is `(A \ B) ⊔ (B \ A)`. -/
+def symm_diff {α : Type*} [has_sup α] [has_sdiff α] (A B : α) : α := (A \ B) ⊔ (B \ A)
+
+infix ` Δ `:100 := symm_diff
+
+@[simp] lemma symm_diff_def {α : Type*} [has_sup α] [has_sdiff α] (A B : α) :
```

Comment:
Are you sure this should be `@[simp]`? That means that the notation `\symm_diff` will *always* be unpacked. I would guess that keeping the notation can make for short and readable goals.

Reply:
```suggestion
lemma symm_diff_def {α : Type*} [has_sup α] [has_sdiff α] (A B : α) :
```